src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,126 @@
+package duke;
+
+import java.io.*;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+public class Storage {
+
+    private static boolean appendMode = true;
+    private static String filePath = "./data/dukeData.txt";
+
+    /**
+     * loads data about the task list from dukeData.txt.
+     * @return an ArrayList of TaskItems that will be used to create Duke's TaskList.
+     */
+    public static ArrayList<TaskItem> loadData() {
```

Comment:
Your `loadData()` method is rather long and deeply nested. Abstracting out certain parts to helper methods/classes could make it simpler!